packages/terra-pills/src/Pills.jsx

```diff
@@ -0,0 +1,313 @@
+import React, {
+  useCallback, useEffect, useLayoutEffect, useRef, useState,
+} from 'react';
+import {
+  KEY_BACK_SPACE, KEY_DELETE, KEY_LEFT, KEY_RIGHT,
+} from 'keycode-js';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import { injectIntl } from 'react-intl';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import Pill from './Private/_Pill';
+import styles from './Private/Pill.module.scss';
+import RollUpPill from './Private/_RollupPill';
+import RollUpUtils from './Private/RollUpUtils';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The string that labels the collection of pills, used in cased where the text label is not visible on the screen and required for minimum accessibility standards. Providing this prop adds the aria-label attribute to the Pills container element. (Required)
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * If additional visible information text is used, provide a string containing the IDs for html elements that help describe the intent of the group of pills.
+   */
+  ariaDescribedBy: PropTypes.string,
+  /**
+   * If a visible text label is used with the collection of pills, provide a string of the ID for the html element containing the visible text label. Supplying the 'ariaLabelledBy' prop will override the 'ariaLabel' prop and adds the aria-labelledby attribute instead to the Pills container element. For best practices, ensure the visible text in the html element id provided to the Pills ariaLabelledbyprop matches the text provided to the ariaLabel prop, for consistency in the case of fallback or errors.
+   */
+  ariaLabelledBy: PropTypes.string,
+  /**
+   * The content to be shown in the pills container. Should only contain the sub-component <Pills.Pill>.
+   */
+  children: PropTypes.node,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+  /**
+   * Determines if the Pills group should be shown as rolled up or not, limited to a single line of display.
+   */
+  isCollapsed: PropTypes.bool,
```

Comment:
I'm not sure the isCollapsed and onSelectRollUp api's are going to be useful to consumers. I'm not sure when and how consumers would know when to force a tabs to collapse, once they're un collapsed theres no common way to collapse them and they will expand the area where you expand them so they can't easily fit in short containers.

Optionally maybe we could offer two states, single line or 'normal' non collapsed. The normal variant would never collapse and the single line variant would always be a single line but offer a disclosure mechanism to show and modify the collapsed pills.

Or maybe we could scope out collapsed for the first release?

Reply:
@neilpfeiffer , @scottwilmarth  can you guys provide your thoughts on the above comment ?

Reply:
We'll be discussing the API and tech design further today and will post direction of how to address.

cc: @scottwilmarth

Reply:
I talked with @neilpfeiffer, on this topic, I think there was some confusion on what `isCollapsed` meant, so we'd like to change the `isCollapsed` prop to `isSingleLine`. A single line pill container may be collapsed or not depending on how many pills it contains.

Reply:
renamed in this commit 91a9d8766efc2524f8114e0d1f9332318e52b26f